### PR TITLE
fix #310

### LINKED ISSUE #310: Add link to tutorial on using the command line.
### Summary

As discussed in #259 most chapters depend on some experience of using the command line, but we don't have a resource for this.

### What needs to be done?

Find a nice tutorial on using the command line. Add it as a link in the relevant chapter's prerequisites sections.

### Who can help?

Anyone confident with the command line.

### PR DESCRIPTION
**Pull request by jcolomb**
will need to update all reference to 
https://programminghistorian.org/en/lessons/intro-to-bash

with {ref}`command line<rr-overview-resources-commandline>`

### Summary

Fixes #310

### List of changes proposed in this PR (pull-request)

added info + resources on command line that were listed in 310

### What should a reviewer concentrate their feedback on?

- [x] Is the information fitted to the purpose (too little/too much)
- [x] Everything looks ok?

### Acknowledging contributors

- [] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file.
- [X] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/main/README.md#contributors) in the README file: 
maybe:
[r-j-arnold](https://github.com/r-j-arnold)
[tomalrussell](https://github.com/tomalrussell)

### CODE REVIEW

File: book/website/reproducible-research/overview/overview-resources.md [hunk 1]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
 (rr-overview-resources-addmaterial)=
 ## Additional Material
 
+(rr-overview-resources-commandline)=
+### Using the command line
+
+While most of what this book is about do not require the use of the command line, it often makes things actually easier.
```
Comment (Arielle-Bennett) [inline book/website/reproducible-research/overview/overview-resources.md:27]:
```suggestion
While most of what this book is about does not require the use of the command line, it often actually makes things easier. 
```
Could we expand here what "things" is referring to? Just in case someone completely new to the terminal comes across this page and isn't sure

File: book/website/reproducible-research/overview/overview-resources.md [hunk 2]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
 (rr-overview-resources-addmaterial)=
 ## Additional Material
 
+(rr-overview-resources-commandline)=
+### Using the command line
+
+While most of what this book is about do not require the use of the command line, it often makes things actually easier. 
+Even when your visual abilities allows you to use efficiently your mouse, writing commands in the terminal/powershell/command is not that difficult.
```
Comment (Arielle-Bennett) [inline book/website/reproducible-research/overview/overview-resources.md:28]: Suggest we change "not that difficult" to something that explains why its a good idea!

File: book/website/reproducible-research/overview/overview-resources.md [hunk 3]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+You first need to know that linux and OSX use `bash` code natively, while the command line in PC uses a different language.
```
Comment (Arielle-Bennett) [inline book/website/reproducible-research/overview/overview-resources.md:29]: Is it PC or Windows? What is the language that it uses?

**Review by Arielle-Bennett — COMMENTED** [review]
Some comments from me on this :)

**Review by jcolomb — COMMENTED** [review]
thanks Arielle. I hope I replied to all comment.

File: book/website/reproducible-research/overview/overview-resources.md [hunk 4]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
 (rr-overview-resources-addmaterial)=
 ## Additional Material
 
+(rr-overview-resources-commandline)=
+### Using the command line
+
+While most of what this book is about does not require the use of the command line, it often actually makes things easier. 
+Even when your visual abilities allows you to use efficiently your mouse, writing commands in the terminal/powershell/command is not that difficult.
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:28]:
```suggestion
Writing commands in the terminal, PowerShell, or command prompt provides an alternative means of interacting with the computer. It offers several advantages and can be considered a beneficial approach for various reasons:
- By directly entering commands, tasks can be performed efficiently without relying on graphical user interfaces (GUIs) or navigating through multiple menus.
- It provides higher level of flexibility and control as the user can execute complex tasks, automate repetitive actions, and customise commands according to specific needs.
- The terminal is available on various operating systems. Hence, knowing how to navigate and use the command-line interface ensures a consistent approach across different platforms.
- Familiarity with the terminal enhances the user's understanding of the underlying operating system and improves their technical skills. 
- When encountering issues or performing system diagnostics, the command-line utilities and diagnostic commands allow the user to investigate and resolve problems at a deeper level than GUI-based tools.
```

I would rephrase to not limit only to visual abilities as people also use speech output systems (like screen-readers), braille displays, and other assistive technologies to use computers. Also I won't comment whether it is a simple way or not rather, I would highlight what are the advantages of using it.

File: book/website/reproducible-research/overview/overview-resources.md [hunk 5]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+You first need to know that linux and OSX use `bash` code natively, while the command line in PC uses a different language.
+
+Here is a small list of tools and training material to know more :
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:31]:
```suggestion
List of tools and training materials for using the command-line:
```

File: book/website/reproducible-research/overview/overview-resources.md [hunk 6]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+- Game for beginners (to advanced): https://gitlab.com/slackermedia/bashcrawl
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:32]:
```suggestion
- A [game to teach one the basics of using a POSIX (Linux, BSD, UNIX) terminal](https://gitlab.com/slackermedia/bashcrawl) 
```

File: book/website/reproducible-research/overview/overview-resources.md [hunk 7]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+(rr-overview-resources-commandline)=
+### Using the command line
+
+While most of what this book is about does not require the use of the command line, it often actually makes things easier. 
+Even when your visual abilities allows you to use efficiently your mouse, writing commands in the terminal/powershell/command is not that difficult.
+You first need to know that linux and OSX use `bash` code natively, while the command line in PC uses a different language.
+
+Here is a small list of tools and training material to know more :
+- Game for beginners (to advanced): https://gitlab.com/slackermedia/bashcrawl 
+- Carpentry lesson: https://swcarpentry.github.io/shell-novice/
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:33]:
```suggestion
- Carpentries lesson on [The Unix Shell](https://swcarpentry.github.io/shell-novice/)
```

File: book/website/reproducible-research/overview/overview-resources.md [hunk 8]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+- https://www.taniarascia.com/how-to-use-the-command-line-for-apple-macos-and-linux/
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:34]:
```suggestion
- An article on [Command Line Tutorial: Usage in Linux and macOS](https://www.taniarascia.com/how-to-use-the-command-line-for-apple-macos-and-linux/)
```

File: book/website/reproducible-research/overview/overview-resources.md [hunk 9]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+- https://programminghistorian.org/en/lessons/intro-to-bash
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:35]:
```suggestion
- A lesson on [Introduction to the Bash Command Line](https://programminghistorian.org/en/lessons/intro-to-bash)
```

File: book/website/reproducible-research/overview/overview-resources.md [hunk 10]
```diff
@@ -21,6 +21,30 @@ If you want to start learning hands-on practices, we recommend reading the {ref}
...
+
+List of some tools and training materials for using the command-line:
+- A [game to teach one the basics of using a POSIX (Linux, BSD, UNIX) terminal](https://gitlab.com/slackermedia/bashcrawl) 
+- Carpentries lesson on [The Unix Shell](https://swcarpentry.github.io/shell-novice/)
+- An article on [Command Line Tutorial: Usage in Linux and macOS](https://www.taniarascia.com/how-to-use-the-command-line-for-apple-macos-and-linux/)
+- A lesson on [Introduction to the Bash Command Line](https://programminghistorian.org/en/lessons/intro-to-bash)
+
+To ease your journey, here are two tips which should make onboarding much easier:
+
+- Get your terminal at the right folder:
```
Comment (SaranjeetKaur) [inline book/website/reproducible-research/overview/overview-resources.md:39]:
```suggestion
- When working on the terminal please make sure that you are in the correct folder (as called as directory):
```

**Review by SaranjeetKaur — APPROVED** [review]

File: book/website/reproducible-research/overview/overview-resources.md (hunk 4 above)
Comment (SaranjeetKaur) [reply book/website/reproducible-research/overview/overview-resources.md:28]: Makes sense!
